Alliance.Client/Extensions/ExNativeUI/LobbyEquipment/ViewModels/EquipmentSelectionVM.cs

```diff
@@ -519,34 +519,18 @@ public EquipmentSelectionVM(MissionMultiplayerGameModeBaseClient gameMode, Actio
                     {
                         ALHeroClassVM = Classes.FirstOrDefault()?.SubClasses.FirstOrDefault();
                     }
+                    else
+                    {
+                        ALHeroClassVM = InitHeroFromPreviousSelection(initialHeroSelection, ALHeroClassVM);
+                    }
                 }
                 else if (IsGoldEnabled && initialHeroSelection.TroopCost > Gold)
                 {
                     ALHeroClassVM = Classes.FirstOrDefault()?.SubClasses.FirstOrDefault();
                 }
                 else
                 {
-                    foreach (ALHeroClassGroupVM @class in Classes)
-                    {
-                        foreach (ALHeroClassVM subClass in @class.SubClasses)
-                        {
-                            if (subClass.HeroClass == initialHeroSelection)
-                            {
-                                ALHeroClassVM = subClass;
-                                break;
-                            }
-                        }
-
-                        if (ALHeroClassVM != null)
-                        {
-                            break;
-                        }
-                    }
-
-                    if (ALHeroClassVM == null)
-                    {
-                        ALHeroClassVM = Classes.FirstOrDefault()?.SubClasses.FirstOrDefault();
-                    }
+                    ALHeroClassVM = InitHeroFromPreviousSelection(initialHeroSelection, ALHeroClassVM);
                 }
```

Comment:
This method `EquipmentSelectionVM` is becoming really big.
I would recommand to split elements in it into small functions in order to make it more readable.